lib/streamlit/elements/widgets/time_widgets.py

```diff
@@ -61,8 +61,21 @@
 if TYPE_CHECKING:
     from streamlit.delta_generator import DeltaGenerator
 
-SingleDateValue: TypeAlias = Union[date, datetime, None]
-DateValue: TypeAlias = Union[SingleDateValue, Sequence[SingleDateValue]]
+# Type for things that point to a specific time.
+TimeValue = Union[time, datetime, str, Literal["now"]]
```

Comment:
nit: I think instead of `Union` we nowadays use the `|` syntax for any new code, so perhaps worth updating this

Reply:
I think the new `|` syntax might not work if used for a type alias.

Reply:
That's what I had thought too, but then I tested it on Python 3.9 and it seemed to work, so I gave it a shot. (Python 3.8 didn't work on my machine, but for other reasons -- some dynamic linking issue)

But looks like tests are failing because of this, so I'm reverting the change now.